src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const stateClone = { ...state };
```

Comment:
[NAMING] - The name `stateClone` is a bit misleading as it implies a single clone, but it is actually being reused across all actions. It would be better to name it something like `currentState` to reflect that it represents the current state after each action.